scanner_test.go

```diff
@@ -46,6 +46,22 @@ func TestScannerLongLine(t *testing.T) {
 	}
 }
 
+func TestLargePayload(t *testing.T) {
+
+	ctx := context.Background()
+	payload := `{"msg":` + strings.Repeat("a", 1024*1024) + `}` // more than 1mb long json payload
+	src := strings.NewReader(payload)
+
+	opts := DefaultOptions()
+	opts.timeNow = func() time.Time {
+		return time.Date(2024, 11, 1, 15, 40, 0, 0, time.UTC)
+	}
+
+	sink := bufsink.NewSizedBufferedSink(1024*1024, nil)
```

Comment:
this sized buffered sink isn't related to the scanning buffer, its simply buffering `[100]*types.LogEvent`
```suggestion
	sink := bufsink.NewSizedBufferedSink(100, nil)
```

in fact this test here should inspect the content of that buffer after parsing a message that's too long:

```
	require.Len(t, sink.Buffered, len(want))
	for i, got := range sink.Buffered {
		require.Equal(t, pjson(want[i]), pjson(got))
	}
```

i think you want to demonstrate 2 things:
- [ ] the scanner didn't die because of the excessively long input
- [ ] the scanner kept on scanning log events despite the input that was too long
right now the test doesn't demonstrate this, afaict